### PR TITLE
Update calendar to close when clicking outside

### PR DESCRIPTION
**Pull request by busypeoples**
### What this PR does

Ensure that the calendar closes when clicking outside

### Checklist:

- [ ] I reviewed my own code
- [ ] The changes align with the designs I received  
  Or give a reason why this does not apply:
- [ ] I have attached screenshot(s), video(s) or gif(s) showing that the solution is working as expected  
  Or give a reason why this does not apply:
- [ ] I have updated the task(s) status on Linear
- [ ] All new media is optimized (images, gifs, videos)

### Browser support

My code works in the following browsers:

- [ ] Firefox
- [ ] Chrome
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: src/lib/components/dateSelector/index.tsx [hunk 1]
```diff
@@ -111,6 +113,22 @@ const DateSelector = ({
...
+        setOpenCalendar(false);
+      }
+    };
+    document.addEventListener('mousedown', handleOutsideClick);
+
+    return () => {
+      document.removeEventListener('mousedown', handleOutsideClick);
+    };
+  }, [calendarContainerRef, setOpenCalendar]);
+
```
Comment (diogomateus) [inline src/lib/components/dateSelector/index.tsx:131]: WDYT about moving this to a the utils or hooks folder? Seems like something that might be useful for other cases. We could even export it and take advantage of that in app/website/others repos

Comment (ruizhen88) [reply src/lib/components/dateSelector/index.tsx:131]: Was going to make the same comment!

Comment (busypeoples) [reply src/lib/components/dateSelector/index.tsx:131]: Makes sense.

**Review by ruizhen88 — APPROVED** [review]

File: src/lib/components/dateSelector/index.tsx (hunk 1 above)
Comment (diogomateus) [reply src/lib/components/dateSelector/index.tsx:131]: We could even move that hook to dirty-swan and import it on website, less duplicated code 🧹 🧹

Comment (busypeoples) [reply src/lib/components/dateSelector/index.tsx:131]: Thanks @ruizhen88 @diogomateus for the feedback. Will check and then update.

**Review by talo242 — APPROVED** [review]
Such a nice contribution 🙌🏼

**Review by kimkwanka — APPROVED** [review]

File: src/lib/components/dateSelector/index.tsx (hunk 1 above)
Comment (busypeoples) [reply src/lib/components/dateSelector/index.tsx:118]: Ok converted the onClickOutside into a hook @diogomateus @ruizhen88

Comment (diogomateus) [reply src/lib/components/dateSelector/index.tsx:118]: Looks good! Should we add some tests as well?

Comment (busypeoples) [reply src/lib/components/dateSelector/index.tsx:118]: Ok, added some test @diogomateus

**Pull request merged**